src/geometry.rs

```diff
@@ -1,9 +1,115 @@
 //! Extension's to Smithay's geometry module.
 
-use std::cmp;
+use std::error::Error;
+use std::str::FromStr;
+use std::{cmp, fmt, ops};
 
 use smithay::utils::{Point, Size};
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Matrix3x3<T: Copy> {
+    storage: Vec<T>,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct FromVecError(usize, usize);
+
+impl fmt::Display for FromVecError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "Mismatched size when createing object from vector, expected {} got {}",
+            self.0, self.1
+        )
+    }
+}
+
+impl Error for FromVecError {}
+
+impl<T: Copy> TryFrom<Vec<T>> for Matrix3x3<T> {
+    type Error = Box<dyn Error>;
+
+    fn try_from(storage: Vec<T>) -> Result<Self, Self::Error> {
+        if storage.len() != 9 {
+            Err(Box::new(FromVecError(9, storage.len())))
+        } else {
+            Ok(Self { storage })
+        }
+    }
+}
```

Comment:
Is there any benefit to a dedicated error type just to wrap it in a box and throw it away?

I feel like at that point we can save us the boilerplate and just return a boxed string.

Reply:
I've tried, and we can't do that. It requires something that is `Box<dyn Error>`, but when you throw a string you we get `Box<dyn String>`. If you know how to approach this would be nice.

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5e550ecbc64fd5e5e4bb44b8ea9c6344

This is the shortest variation, but generally with `Box<dyn _>` you can just always do `Box::new()` and then `as Box<dyn Whatever>` iirc.

Reply:
Hm, I think I've tried putting inside box before hand originally or something like that. Thx.